packages/flutter/lib/src/material/segmented_button.dart

```diff
@@ -835,7 +872,33 @@ class _RenderSegmentedButton<T> extends RenderBox with
     return Size(childWidth, maxHeight);
   }
 
+  Size _calculateVerticalChildSize(BoxConstraints constraints) {
+    double maxWidth = 0;
+    RenderBox? child = firstChild;
+    double childHeight;
+    if (_isExpanded) {
+      childHeight = constraints.maxHeight / childCount;
+    } else {
+      childHeight = constraints.minHeight / childCount;
+      while (child != null) {
+        childHeight = math.max(childHeight, child.getMaxIntrinsicHeight(double.infinity));
+        child = childAfter(child);
+      }
+      childHeight = math.min(childHeight, constraints.maxWidth / childCount);
+    }
+    child = firstChild;
+    while (child != null) {
+      final double boxHeight = child.getMaxIntrinsicWidth(maxWidth);
```

Comment:
This variable name is a little confusing. Would be better if we can change it to `boxWidth`:)

Reply:
I agree :)